examples/filters/colorfilter/color_filter.go

```diff
@@ -0,0 +1,152 @@
+// Package colorfilter implements a modular camera that filters the output of an underlying camera and only keeps
+// captured data if the vision service detects a certain color in the captured image.
+package colorfilter
+
+import (
+	"context"
+	"fmt"
+	"image"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"github.com/viamrobotics/gostream"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/services/vision"
+)
+
+var (
+	Model            = resource.NewModel("example", "filter", "colorfilter")
+	errUnimplemented = errors.New("unimplemented")
+)
+
+func init() {
+	resource.RegisterComponent(camera.API, Model, resource.Registration[camera.Camera, *Config]{
+		Constructor: newCamera,
+	})
+}
+
+func newCamera(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (camera.Camera, error) {
+	c := &colorFilterCam{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+	if err := c.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+type Config struct {
+	ActualCam     string `json:"actual_cam"`
+	VisionService string `json:"vision_service"`
+}
+
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.ActualCam == "" {
+		return nil, fmt.Errorf(`expected "actual_cam" attribute in %q`, path)
+	}
+
+	return []string{cfg.ActualCam}, nil
+}
+
+type colorFilterCam struct {
```

Comment:
Let's add a comment above this type so any users reading this example will know what it does!

Reply:
Added!